x-pack/plugins/apm/server/lib/errors/get_error_rate.ts

```diff
@@ -30,80 +33,49 @@ export async function getErrorRate({
 
   const filter = [
     { term: { [SERVICE_NAME]: serviceName } },
+    { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
     { range: rangeFilter(start, end) },
     ...uiFiltersES,
   ];
 
-  const aggs = {
-    response_times: {
-      date_histogram: getMetricsDateHistogramParams(start, end),
-    },
-  };
-
-  const getTransactionBucketAggregation = async () => {
-    const resp = await client.search({
-      index: indices['apm_oss.transactionIndices'],
-      body: {
-        size: 0,
-        query: {
-          bool: {
-            filter: [
-              ...filter,
-              { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
-            ],
-          },
-        },
-        aggs,
+  const params = {
+    index: indices['apm_oss.transactionIndices'],
+    body: {
+      size: 0,
+      query: {
+        bool: { filter },
       },
-    });
-    return {
-      totalHits: resp.hits.total.value,
-      responseTimeBuckets: resp.aggregations?.response_times.buckets,
-    };
-  };
-  const getErrorBucketAggregation = async () => {
-    const groupIdFilter = groupId
-      ? [{ term: { [ERROR_GROUP_ID]: groupId } }]
-      : [];
-    const resp = await client.search({
-      index: indices['apm_oss.errorIndices'],
-      body: {
-        size: 0,
-        query: {
-          bool: {
-            filter: [
-              ...filter,
-              ...groupIdFilter,
-              { term: { [PROCESSOR_EVENT]: ProcessorEvent.error } },
-            ],
+      aggs: {
+        histogram: {
+          date_histogram: getMetricsDateHistogramParams(start, end),
+          aggs: {
+            statusAggregation: {
+              terms: {
+                field: HTTP_RESPONSE_STATUS_CODE,
+                size: 10,
```

Comment:
This is not going to work if there are more than 10 buckets. A histogram with a bucket size of 100 will be better.
Another approach that I think is even simpler is go make two ES requests. One gets the total number of transactions in the time range. The second gets the total number of erroneous transactions. Then you can simply divide those two numbers.

Reply:
Yeah, that sounds better indeed. Thanks for the feedback.

Reply:
@sqren do you for instance have an idea on how to fetch only the erroneous transactions? I can do a terms filter, but I'd have to manually pass all possible status in the array.
```
 {
          "terms": {
            "http.response.status_code": ["500", "400"...]
          }
        }
```

Reply:
You can use a [range query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html):

```jsonc
    "query": {
        "range" : {
            "http.response.status_code" : {
                "gte" : 400 // everything equals or above 400 should be treated as an error
            }
        }
    }
```

Reply:
I now see you already figured this out with @dgieselaar 👍

Reply:
Yes, thanks for that anyway.